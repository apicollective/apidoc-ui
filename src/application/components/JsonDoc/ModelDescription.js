// @flow
import React from 'react';

import H2 from '../../../components/H2';
import Markdown from '../../../components/Markdown';

import type { Service } from '../../../generated/version/ServiceType';

import * as utils from '../../../utils';
import styles from './json-doc.css';

const ModelDescription = ({
  baseModel,
  service,
  importedServices,
  modelNameClick,
}: {
  baseModel: string,
  service: Service,
  importedServices: Service[],
  modelNameClick: (event: Object) => void,
}) => {
  const type = utils.getType(baseModel);
  const model = utils.getModel(type, service, importedServices);

  //   <H2 click={model ? this.props.modelNameClick : null} className={styles.modelName}>
  //   <H2 className={styles.modelName}>
  return (
    <div className={styles.modelDescription}>
      <H2 className={styles.modelName}>
        {utils.simplifyName(baseModel)}
      </H2>
      {model && model.description
        ? <Markdown
            source={model.description ? model.description : ''}
            className={styles.description}
          />
        : null}
    </div>
  );
};

export default ModelDescription;
