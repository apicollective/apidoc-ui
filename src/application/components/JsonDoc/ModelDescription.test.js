// @flow
import React from 'react';
import renderer from 'react-test-renderer';
/* import { test, expect } from 'jest';*/

import ModelDescription from './ModelDescription';
import Example from '../../../exampleService.json';

test('MD', () => {
  const params = {
    baseModel: 'base',
    service: Example,
    importedServices: [],
    modelNameClick: () => {},
  };

  const component = renderer.create(
    <ModelDescription {...params} />,
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
