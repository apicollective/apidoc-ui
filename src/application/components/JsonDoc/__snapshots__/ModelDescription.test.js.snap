exports[`test MD 1`] = `
<div
  className="modelDescription">
  <h2
    className="modelName h2"
    onClick={undefined}>
    base
  </h2>
</div>
`;
